### PR TITLE
mkdocs: 1.2.1 -> 1.2.2 pymdown-extensions: init at 8.2 mkdocs-materia…

### PR DESCRIPTION
**Pull request by lde**
mkdocs: 1.2.1 -> 1.2.2
pymdown-extensions: init at 8.2
mkdocs-material-extensions: init at 1.0.3
mkdocs-material: init at 7.2.8

adding mkdocs-material and its deps.

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/documentation/mkdocs/default.nix [hunk 1]
```diff
@@ -30,26 +29,11 @@ buildPythonApplication rec {
...
-  pytestFlagsArray = [ "mkdocs/tests/*.py" ];
-
-  disabledTests = [
-    # Don't start a test server
-    "testing_server"
-  ];
-
   pythonImportsCheck = [ "mkdocs" ];
 
+  doCheck = false;
```
Comment (fabaff) [inline pkgs/development/tools/documentation/mkdocs/default.nix:34]: It should still be possible to run the tests even if upstream moved to `tox`.

```bash
  checkPhase = ''
    ${python.interpreter} -m unittest discover -p "*tests.py" mkdocs
  '';
```

**Review by fabaff — COMMENTED** [review]
Please use a separate commit for every package in this PR.

File: pkgs/development/python-modules/mkdocs-material-extensions/default.nix [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  pname = "mkdocs-material-extensions";
+  version = "1.0.3";
+  disabled = pythonOlder "3.6";
+
+  src = fetchFromGitHub {
+        repo = pname;
+        owner = "facelessuser";
+        rev = version;
+        sha256 = "1mvc13lz16apnli2qcqf0dvlm8mshy47jmz2vp72lja6x8jfq2p3";
+      };
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/mkdocs-material-extensions/default.nix:17]:
```suggestion
  src = fetchFromGitHub {
    repo = pname;
    owner = "facelessuser";
    rev = version;
    sha256 = "1mvc13lz16apnli2qcqf0dvlm8mshy47jmz2vp72lja6x8jfq2p3";
  };
```

File: pkgs/development/python-modules/mkdocs-material/default.nix [hunk 3]
```diff
@@ -0,0 +1,45 @@
+{ lib
+, buildPythonApplication
+, fetchFromGitHub
+, pythonOlder
+, python3
+, mkdocs
+}:
+
+with python3.pkgs;
+
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/mkdocs-material/default.nix:10]:
```suggestion
```
This is not allowed. Please list the packages as individual inputs.

File: pkgs/development/python-modules/pymdown-extensions/default.nix [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+  propagatedBuildInputs = [
+    markdown
+  ];
+
+  # test fails due to missing deps
+  doCheck = false;
+
+  pythonImportsCheck = [ "pymdownx" ];
+
+
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pymdown-extensions/default.nix:26]:
```suggestion

```

File: pkgs/development/tools/documentation/mkdocs/default.nix [hunk 5]
```diff
@@ -7,16 +7,15 @@ with python3.pkgs;
 
 buildPythonApplication rec {
   pname = "mkdocs";
-  version = "1.2.1";
   disabled = pythonOlder "3.6";
+  version = "1.2.2";
```
Comment (SuperSandro2000) [inline pkgs/development/tools/documentation/mkdocs/default.nix:11]: Please revert that.

File: pkgs/development/tools/documentation/mkdocs/default.nix [hunk 6]
```diff
@@ -30,26 +29,11 @@ buildPythonApplication rec {
...
-
-  disabledTests = [
-    # Don't start a test server
-    "testing_server"
-  ];
-
   pythonImportsCheck = [ "mkdocs" ];
 
+  doCheck = false;
+  # tests mirgated to tox
```
Comment (SuperSandro2000) [inline pkgs/development/tools/documentation/mkdocs/default.nix:35]: Comments belong above the lines thy are about.

**Review by jonringer — COMMENTED** [review]
To comply with [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#submitting-changes) please have the commit message name be of the format
```
<pkg-name>: <subject-line>
```
for more examples, please look at https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#submitting-changes

in your case, please have a commit per new package

**Pull request closed without merging**